File: internal/core/scope.go
package core

import (
	"fmt"
	"strings"
)

const (
	ScopeGlobal  = "global"
	ScopeProject = "project"
)

// NormalizeScope canonicalises a scope string. Empty input maps to
// ScopeGlobal so callers that don't care about scope (back-compat paths)
// land in the global namespace by default.
func NormalizeScope(s string) (string, error) {
	switch strings.ToLower(strings.TrimSpace(s)) {
	case "", ScopeGlobal:
		return ScopeGlobal, nil
	case ScopeProject:
		return ScopeProject, nil
	default:
		return "", fmt.Errorf("unknown scope %q (expected %q or %q)", s, ScopeGlobal, ScopeProject)
	}
}

File: knowledger/core/scope.py
SCOPE_GLOBAL = "global"
SCOPE_PROJECT = "project"


def normalize_scope(s: str) -> str:
    normalized = s.strip().lower()
    if normalized in ("", SCOPE_GLOBAL):
        return SCOPE_GLOBAL
    if normalized == SCOPE_PROJECT:
        return SCOPE_PROJECT
    raise ValueError(f"unknown scope {s!r} (expected {SCOPE_GLOBAL!r} or {SCOPE_PROJECT!r})")
